dbt_semantic_interfaces/implementations/saved_query.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from typing import List, Optional, Sequence
+
+from typing_extensions import override
+
+from dbt_semantic_interfaces.implementations.base import (
+    HashableBaseModel,
+    ModelWithMetadataParsing,
+)
+from dbt_semantic_interfaces.implementations.filters.where_filter import (
+    PydanticWhereFilter,
+)
+from dbt_semantic_interfaces.implementations.metadata import PydanticMetadata
+from dbt_semantic_interfaces.protocols import ProtocolHint
+from dbt_semantic_interfaces.protocols.saved_query import SavedQuery
+
+
+class PydanticSavedQuery(HashableBaseModel, ModelWithMetadataParsing, ProtocolHint[SavedQuery]):
+    """Pydantic implementation of SavedQuery."""
+
+    @override
+    def _implements_protocol(self) -> SavedQuery:
+        return self
+
+    name: str
+    metrics: List[str]
+    group_bys: List[str]
```

Comment:
@QMalcolm does it matter that this is not part of the protocol? It shouldn't, but I wanted to make sure this won't cause havoc with the core parsing setup.

Reply:
Good catch. It would make things weird. To stick to the protocol we'd have to serialize `group_by_item_names`, but then MetricFlow uses these pydantic objects for deserialization, and since their is no setter for `group_by_item_names` on this object, they'd get dropped 😬 In a separate comment you brought up a question about why do we have both `group_bys` and `group_by_item_names`, and I think that's actually the better question. I think we just want `group_bys`? I'm not seeing what `group_by_item_names` provides.